src/main/java/tc/oc/pgm/match/MatchManagerImpl.java

```diff
@@ -157,7 +157,7 @@ public void unloadMatch(@Nullable String id) {
       match.unload();
     }
 
-    matchIdByWorldName.remove(match.getWorld().getName());
+    matchIdByWorldName.remove("match-" + id);
```

Comment:
There should be a separate method to format world names, used by this and the `WorldCreator`.